clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1689,10 +1689,10 @@ bool CompilerInvocation::ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args,
         {std::string(Split.first), std::string(Split.second)});
   }
 
-  Opts.DisableLLVMPasses =
-      Args.hasArg(OPT_disable_llvm_passes) ||
-      (Args.hasArg(OPT_fsycl_is_device) && T.isSPIR() &&
-       Args.hasArg(OPT_fno_sycl_early_optimizations));
+  Opts.DisableLLVMPasses = Args.hasArg(OPT_disable_llvm_passes);
+
+  Opts.DisableSYCLEarlyOpts = Args.hasArg(OPT_fsycl_is_device) && T.isSPIR() &&
```

Comment:
Do we really need to do these checks? Can't we just set this via marshalling infrastructure if option is passed?

Reply:
I don't believe this is needed and can be accomplished just with the marshalling facilities as you suggest. All the tests pass without this, and both of these checks are already used elsewhere in the `RunOptimizationPipeline` function which is the main area of work for the patch.

If it's better to remove this and just have the definition in `Options.td`, I'll remove this.

Reply:
Yes please do